docs/proposal-spawn-on-d.md

```diff
@@ -56,46 +58,51 @@ and usage, and will make Cylc 9 changes easier to implement in the future.
-  runahead pool then released as the workflow moves forward)
+- *runahead pool*: task proxies that have been spawned (created) already but
+  are beyond the current runahead limit, or beyond the (non-final) stop point.
+  All tasks are initially spawned into the runahead pool then released to the
+  main pool as the worklow (and the runahead limit) moves foreward.
 - *n=0 window*: tasks that anchor the current view (by default, the content of
   the active task pool)
 - *n=M window*: tasks within M graph edges of the those in the n=0 window
-- *finished*: means *succeeded*, OR *failed* with ":fail" handled
-- *reflow*: when the workflow flows on from a manually triggered task
+- *finished*: means *succeeded*, OR *failed* with ":fail" handled. An unhandled
+  failed task is consider to be "not finished"
+- *reflow*: when the workflow flows on according to the graph, from a manually
+  triggered task
 
 ## Implementation overview
 
-1. Get tasks to record (from the graph) the children that depend on each of
-      their outputs 
-1. Continually [auto-spawn tasks with no
-      parents](#spawning-tasks-with-no-parents) to the runahead limit
-1. (Tasks with no prerequisites can submit their jobs, as usual)
-1. As outputs are completed [spawn waiting
+1. During graph parsing, record (in parent task definitions) the children that
+     depend on each output
+1. At start-up, spawn the first instance of any [task with no
+        parents](#spawning-tasks-with-no-parents) to the runahead pool 
+    - Then whenever such a task is released from the runahead pool, spawn its
+      next instance (into the runahead pool)
+1. Tasks with all prerequisites satisfied (in the main task pool) can submit
+      their jobs
+1. As task job outputs are completed [spawn
      children](#spawning-waiting-tasks-on-outputs) that depend on them
     (if not already spawned) and update their prerequisites directly
-1. (Tasks with prerequisites all satisfied can submit their jobs, as usual)
-1. As tasks finish, spawn remaining children (if not already spawned) and
+1. As tasks finish, spawn any remaining children (if not already spawned) and
       update their [parent-is-finished
       status](spawning-and-parent-is-finished-status) directly
-1. [Remove waiting tasks](#housekeeping-waiting-and-finished-tasks) if their
-parents have finished
-1. [Remove finished tasks](#housekeeping-waiting-and-finished-tasks) if their
-parents have finished *or* the pool has moved beyond the point it can affect them
-1. [Shut down](#scheduler-shutdown) if stalled with no failed tasks present
+1. [Remove waiting tasks](#housekeeping-waiting-and-finished-tasks) if all of
+      their parents have finished
+1. [Remove finished tasks](#housekeeping-waiting-and-finished-tasks) if all of
+      their parents have finished *or* if the pool has moved beyond the point
+      it can affect them
+1. [Shut down](#stall-and-shutdown) if stalled with no failed tasks present
 
 ## Discussion of details
```

Comment:
`(and maybe [unhandled succeeded  tasks](#succeeded-tasks)?)`
This bit (in the description of the task pool below) is now a broken link (and we've now said no to this anyway).

Reply:
(Removed that broken bit).